linkerd/app/outbound/src/lib.rs

```diff
@@ -119,15 +120,12 @@ impl<S> Outbound<S> {
 }
 
 impl Outbound<()> {
-    pub fn serve<B, P, R>(
-        self,
-        bind: B,
-        profiles: P,
-        resolve: R,
-    ) -> (Local<ServerAddr>, impl Future<Output = ()>)
+    pub async fn serve<L, A, I, P, R>(self, listen: L, profiles: P, resolve: R)
     where
-        B: Bind<ServerConfig>,
-        B::Addrs: Param<Remote<ClientAddr>> + Param<OrigDstAddr>,
+        L: Stream<Item = io::Result<(A, I)>> + Send + Sync + 'static,
```

Comment:
nit/tioli: similarly, since the `L` type param doesn't appear in any other `where` bounds, seems like it could be `impl Stream`:
```rust
    pub async fn serve<A, I, P, R>(
        self,
        listen: impl Stream<Item = io::Result<(A, I)>> + Send + Sync + 'static,
        profiles: P,
        resolve: R,
    ) where
```